docs/proposals/19-migrating-to-prometheus-operator.md.md

```diff
@@ -138,60 +170,88 @@ out in the following steps:
           version: v2.35.0
         ```
 
-        - Prometheus will only discover `*Monitors` in its own namespace. This
-          is done by not specifiying `prometheus.spec.*MonitorSelector`.
+    1. Contract between the shoot `Prometheus` and its configuration.
+
+      - `Prometheus` can discover `*Monitors` in different namespaces and also
+        by using labels.
+
+      - Since most of the monitoring configuration is always the same, each
+        Prometheus can reuse configuration which is just generated once. This
+        configuration can be stored in a `monitoring` namespace. To make sure
+        Prometheus will only scrape targets in its own namespace,
+        `ignoreNamespaceSelectors=true` must be set.
+
+      - In some cases, specific configuration is required (e.g. specific
+        configuration due to K8s versions). In this case, the configuration
+        should be deployed in the shoot's namespace and Prometheus will also be
+        able to discover this configuration.
+
+      - To discover `ServiceMonitors` in the `monitoring` namespace and in the
+        control plane of the shoot, the `serviceMonitorNamespaceSelector` field
+        must look like this:
+
+        ```yaml
+        serviceMonitorNamespaceSelector:
+          matchExpressions:
+          - key: kubernetes.io/metadata.name
+            operator: In
+            values:
+            - monitoring
+            - shoot--project--name
+        ```
 
-        - In addition to a Prometheus, the configuration must also be created. To
-          do this, each `job` in the prometheus configuration will need to be
-          replaced with either a `ServiceMonitor`, `PodMonitor`, or `Probe`.
-          This `ServiceMonitor` would be picked up by the prometheus defined in
-          the previous step and scrape any service in the namespace
-          `shoot--project--name` that has the label `app=prometheus` on the port
-          called `metrics`.
+      - In addition to discovering `*Monitors` in different namespaces,
+        Prometheus must also distiguish between `*Monitors` relevant for seed
+        targets and shoot targets. This can be done with a
+        `serviceMonitorSelector` and `podMonitorSelector` where `target=seed`.
+        For a `ServiceMonitor` it would look like this:
+
+        ```yaml
+        serviceMonitorSelector:
+          matchLabels:
+            gardener.cloud/monitoring-target: seed
```

Comment:
Should we use `monitoring.gardener.cloud/` as a prefix for all well-known labels in this proposal?